### PR TITLE
OBDS-99 Added organization and locationGroup modal

### PR DESCRIPTION
**Pull request by MagdaToczek**

### CODE REVIEW

File: src/js/components/locations-configuration/LocationDetails.jsx [hunk 1]
```diff
@@ -207,41 +265,60 @@ class LocationDetails extends Component {
   render() {
     return (
       <div className="configuration-wizard-content flex-column">
+        <AddOrganizationModal
```
Comment (pmuchowski) [inline src/js/components/locations-configuration/LocationDetails.jsx:268]: move this inside the form, so you will have access to the mutators it, can be before `<form`, but inside `<Form`

File: src/js/components/locations-configuration/LocationDetails.jsx [hunk 2]
```diff
@@ -207,41 +265,60 @@ class LocationDetails extends Component {
...
-          render={({ form, handleSubmit, values }) => (
-            <form onSubmit={handleSubmit} className="w-100">
-              <div className="classic-form with-description">
+          mutators={{
+            setValue: ([field, value], state, { changeValue }) => {
+              changeValue(state, field, () => value);
+            },
+          }}
+          render={({ form, handleSubmit, values }) => {
+            if (!window.setFormValue) window.setFormValue = form.mutators.setValue;
```
Comment (pmuchowski) [inline src/js/components/locations-configuration/LocationDetails.jsx:288]: remove this, it's not how it should be done

File: src/js/components/locations-configuration/LocationDetails.jsx [hunk 3]
```diff
@@ -207,41 +265,60 @@ class LocationDetails extends Component {
   render() {
     return (
       <div className="configuration-wizard-content flex-column">
+        <AddOrganizationModal
+          isOpen={this.state.showOrganizationModal}
+          onClose={() => this.setState({ showOrganizationModal: false })}
+          onResponse={this.setOrganization}
```
Comment (pmuchowski) [inline src/js/components/locations-configuration/LocationDetails.jsx:271]: when you move this, you will be able to use here `form.mutators.changeOrganization`

**Review by pmuchowski — CHANGES_REQUESTED** [review]

File: src/js/components/locations-configuration/LocationDetails.jsx [hunk 4]
```diff
@@ -156,11 +166,59 @@ class LocationDetails extends Component {
...
-    // eslint-disable-next-line no-console
-    console.log('new organization modal open');
+    this.setState({ showOrganizationModal: true });
+  }
+
+  openNewLocationGroupModal() {
+    this.setState({ showLocationGroupModal: true });
+  }
+
+  fetchOrganization() {
```
[the remark this answers is not in the document]
Comment (MagdaToczek) [reply src/js/components/locations-configuration/LocationDetails.jsx:199]: In response is only id of organization, that is why I want to fetch it.

File: grails-app/i18n/messages.properties [hunk 5]
```diff
@@ -3450,6 +3458,7 @@ react.locationsConfiguration.forecast.expectedLeadTime.tooltip.label=Enter the a
 react.locationsConfiguration.forecast.replenishmentPeriod.tooltip.label=Enter the time period for which you place replenishment orders. For example, if this depot always requests stock for a month, enter 30 days.
 react.locationsConfiguration.forecast.demandTime.tooltip.label=Enter how many days of consumption history you want to use to calculate average demand. For example, to calculate demand using data for the last 3 months, enter 90. For the last 12 months, enter 360.
 react.locationsConfiguration.forecast.notsupported.label=This section is not supported for the chosen Location Type.
+<<<<<<< HEAD
```
Comment (pmuchowski) [inline grails-app/i18n/messages.properties:3461]: fix the conflicts

File: src/js/components/locations-configuration/LocationDetails.jsx [hunk 6]
```diff
@@ -279,117 +290,141 @@ class LocationDetails extends Component {
...
+                  </div>
+
                   <SelectField
-                    fieldName="supportedActivities"
+                    fieldName="locationType"
                     fieldConfig={{
+                      label: 'react.locationsConfiguration.locationType.label',
+                      defaultMessage: 'Location Type',
                       attributes: {
                         multi: true,
```
Comment (pmuchowski) [inline src/js/components/locations-configuration/LocationDetails.jsx:360]: why you added the multi here?

File: src/js/components/locations-configuration/modals/AddOrganizationModal.jsx [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+      rows: 3,
+      withTooltip: true,
+      tooltip: 'react.locationsConfiguration.organizationDescription.tooltip.label',
+    },
+  },
+};
+
+class AddOrganizationModal extends Component {
+  componentDidMount() {
+    this.props.fetchTranslations('', 'locationsConfiguration');
```
Comment (pmuchowski) [inline src/js/components/locations-configuration/modals/AddOrganizationModal.jsx:56]: is this needed here?

File: src/js/components/locations-configuration/modals/AddOrganizationModal.jsx [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.locale && this.props.locale !== nextProps.locale) {
```
Comment (pmuchowski) [inline src/js/components/locations-configuration/modals/AddOrganizationModal.jsx:60]: as above

File: src/js/components/locations-configuration/modals/AddLocationGroupModal.jsx [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+      required: true,
+      withTooltip: true,
+      tooltip: 'react.locationsConfiguration.locationGroupName.tooltip.label',
+    },
+  },
+};
+
+class AddLocationGroupModal extends Component {
+  componentDidMount() {
+    this.props.fetchTranslations('', 'locationsConfiguration');
```
Comment (pmuchowski) [inline src/js/components/locations-configuration/modals/AddLocationGroupModal.jsx:45]: is this needed?

File: src/js/components/locations-configuration/LocationDetails.jsx [hunk 10]
```diff
@@ -279,117 +290,141 @@ class LocationDetails extends Component {
           validate={validate}
           initialValues={this.state.values}
           mutators={{
+            setFieldValue: ([{ id, name, fieldName }], state, utils) => {
```
Comment (pmuchowski) [inline src/js/components/locations-configuration/LocationDetails.jsx:293]: add two separate methods, instead of having fieldName as a parameter

**Review by pmuchowski — CHANGES_REQUESTED** [review]

File: src/js/components/locations-configuration/LocationDetails.jsx (hunk 6 above)
Comment (DaRoTP) [reply src/js/components/locations-configuration/LocationDetails.jsx]: this was probably a mistake during conflict resolution

File: src/js/utils/option-utils.jsx [hunk 11]
```diff
@@ -155,6 +155,24 @@ export const debounceOrganizationsFetch = (waitTime, minSearchLength) =>
     }
   }, waitTime);
 
+export const debounceAllOrganizationsFetch = (waitTime, minSearchLength) =>
+  _.debounce((searchTerm, callback) => {
+    if (searchTerm && searchTerm.length >= minSearchLength) {
+      apiClient.get(`/openboxes/api/organizations?q=${searchTerm}`)
```
Comment (awalkowiak) [inline src/js/utils/option-utils.jsx:161]: I think you should parametrize roleType in the debouncedOrganizationFetch. Do it like `activityCodes` in `debouncedOrganizationFetch`, which by default will be `['ROLE_SUPPLIER']` and for the case when you want a all organization, then you can pass empty list.

**Review by awalkowiak — CHANGES_REQUESTED** [review]

File: src/js/utils/option-utils.jsx [hunk 12]
```diff
@@ -137,10 +137,15 @@ export const debounceProductsInOrders = (waitTime, minSearchLength, vendor, dest
...
 
-export const debounceOrganizationsFetch = (waitTime, minSearchLength) =>
+export const debounceOrganizationsFetch = (waitTime, minSearchLength, roleTypes = ['ROLE_SUPPLIER']) =>
   _.debounce((searchTerm, callback) => {
     if (searchTerm && searchTerm.length >= minSearchLength) {
-      apiClient.get(`/openboxes/api/organizations?q=${searchTerm}&roleType=ROLE_SUPPLIER`)
+      apiClient.get('/openboxes/api/organizations', {
+        params: {
+          q: searchTerm,
+          roleType: roleTypes.length > 0 ? roleTypes.join(',') : undefined,
```
Comment (awalkowiak) [inline src/js/utils/option-utils.jsx:146]: I think this won't work if there will be multiple role types

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**